src/assignment/views/AssignmentDetail.tsx

```diff
@@ -210,6 +225,55 @@ const AssignmentDetail: FC<DefaultSecureRouteProps<{ id: string }>> = ({
 		getRelatedAssignments();
 	}, [getRelatedAssignments]);
 
+	const toggleBookmark = async () => {
+		try {
+			if (!user) {
+				ToastService.danger(
+					tHtml(
+						'collection/views/collection-detail___er-was-een-probleem-met-het-controleren-van-de-ingelogde-gebruiker-log-opnieuw-in-en-probeer-opnieuw'
+					)
+				);
+				return;
+			}
+
+			if (!assignment) {
+				ToastService.danger(
+					tHtml('Er ging iets mis met het ophalen van de ophalen van de opdracht')
+				);
```

Comment:
return after showing error, so the rest of the code is not executed
```suggestion
				);
				return;
```